libraries/AP_SmartRTL/AP_SmartRTL.cpp

```diff
@@ -163,6 +163,25 @@ bool AP_SmartRTL::pop_point(Vector3f& point)
     return true;
 }
 
+// peek at next point on the path without removing it form the path. Returns true on success
+bool AP_SmartRTL::peek_point(Vector3f& point)
+{
+    // check we are active
+    if (!_active) {
+        return false;
+    }
+
+    // check we have another point
+    if (_path_points_count == 0) {
+        return false;
+    }
+
+    // return last point and remove from path
```

Comment:
Well, you're not removing, that's kind of the point.

This still needs to do all of the locking or you have a time-of-check-to-time-of-use bug here.

I suggest you rename pop_point to peek_point , remove the removal part and the reimplement pop_point in terms of peek_point

Reply:
fixed thanks.